File: be/src/test/java/nl/zzpmatcher/userlogon/business/UserRegisterCommandTest.java
package nl.zzpmatcher.userlogon.business;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.junit.Assert.*;

public class UserRegisterCommandTest {

    @Test
    public void createUser() throws Exception {

        final UserRegisterCommand userRegisterCommand = new UserRegisterCommand();
        userRegisterCommand.setEmailaddress("controller@domain.com");
        userRegisterCommand.setPassword("password");

        final User user = userRegisterCommand.createUser();

        assertThat(user.getEmailaddress(), equalTo("controller@domain.com"));
        assertThat(user.getPassword().length(), equalTo(60));
    }

}

File: be/src/test/java/nl/zzpmatcher/userlogon/business/UserRegisterCommandTest.kt
package nl.zzpmatcher.userlogon.business

import org.junit.Test

import org.hamcrest.CoreMatchers.equalTo
import org.junit.Assert.*

class UserRegisterCommandTest {

    @Test
    fun createUser() {

        val userRegisterCommand = UserRegisterCommand()
        userRegisterCommand.emailaddress = "controller@domain.com"
        userRegisterCommand.password = "password"

        val user = userRegisterCommand.createUser()

        assertThat<String>(user.emailaddress, equalTo("controller@domain.com"))
        assertThat(user.password!!.length, equalTo(60))
    }

}